### ISSUE TITLE
podman should not add host.containers.internal for ipvlan networks

### ISSUE THREAD

**Issue by nivekuil** [containers/podman#11351]
/kind bug

**Description**
When connected to an ipvlan, pod's host.containers.internal points to gateway ip, not host ip, and even if it did it wouldn't be routable. As a result `host.containers.internal` is unusable since it always resolves to a broken ip.

**Steps to reproduce the issue:**
If you connect a pod to an ipvlan network, something like this:
```json
{"cniVersion":"0.4.0","name":"cni-podman1","plugins":
[{"ipam":{
"ranges":[[{"rangeStart":"ffff:ffff:100:d6::3","subnet":"ffff:ffff:100::/48"}]],
"routes":[{"dst":"::/0"}],"type":"host-local"},"mode":"l2","type":"ipvlan","master":"eth0"}]}
```
in this case the gateway (as determined by subnet, so `ffff:ffff:100::1`) is different from the host, but podman will point `host.containers.internal` to it anyway. Furthermore host<->container isn't possible through this interface anyway, so podman should just do nothing.

**Comment by rhatdan** [comment]
@Luap99 PTAL

**Comment by Luap99** [comment]
I think I can fix this for 4.0, it should be simple to check for the bridge driver before we add this entry. Fixing this for 3.X would be to much work.
